### ISSUE TITLE
Unhardcode ChallengeGenerals

### ISSUE THREAD

**Issue by tomsons26** [TheAssemblyArmada/Thyme#19]
### Issue explanation
ChallengeGenerals is hardcoded to create 12 GeneralPersona limiting any modders to only be able to mod existing generals instead of adding new ones

**Comment by ghost** [comment]
Also, isn't the Challenge series missing some general specific fights, AFAIK ?

**Comment by mrtnptrs** [comment]
@ZexaronS Uhhhh, what do you mean exactly? :)

**Comment by ghost** [comment]
@mrtnptrs You never go against Demo General in the Challenges with the official game, it's missing.

**Comment by OmniBlade** [comment]
Isn't that an issue with the content being incomplete and not referenced in the control files though? That isn't an engine issue and is unrelated to the fact there is a hard coded limit.

**Comment by ghost** [comment]
@OmniBlade Yeah, that's why I said "Also" ... i meant that stuff like that which is clearly unfinished should be finished or in the case of the engine implemented with more flexibility, ofcourse the ingame logic would be the same, but a modder would have more freedom, where it comes to finishing what was meant to be but wasn't, that would be a separate project ofcourse. I must be thinking so much about all the rest heh.

**Comment by OmniBlade** [comment]
It isn't unfinished as far as the game engine is concerned though, the game as I understand it DOES support having the Demo General in challenges, but it isn't hooked up in the games configuration files. Please don't post comments on issues unless it specifically pertains to the issue at hand which in this case is to increase the number of challenge generals that can exist, not to more fully implement ones that already do exist in the engine.

I've already discussed in IRC that incomplete game features that are only unfinished in terms of configuration data or assets do not fall under the scope of the Thyme project as far as I am concerned.
